sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/implementation/source/CosmosSourceTaskConfig.java

```diff
@@ -0,0 +1,140 @@
+                SOURCE_FEED_RANGE_TASK_UNITS,
+                ConfigDef.Type.STRING,
+                ConfigDef.NO_DEFAULT_VALUE,
+                ConfigDef.Importance.HIGH
+            )
+            .defineInternal(
+                SOURCE_METADATA_TASK_UNIT,
+                ConfigDef.Type.STRING,
+                null,
+                ConfigDef.Importance.HIGH
+            );
+    }
+
+    private List<FeedRangeTaskUnit> parseFeedRangeTaskUnits() {
+        try {
+            String feedRangesTaskUnitsConfig = this.getString(SOURCE_FEED_RANGE_TASK_UNITS);
+            if (!StringUtils.isEmpty(feedRangesTaskUnitsConfig)) {
+                return OBJECT_MAPPER
+                    .readValue(feedRangesTaskUnitsConfig, new TypeReference<List<String>>() {})
+                    .stream()
+                    .map(taskUnitConfigJson -> {
+                        try {
+                            return OBJECT_MAPPER.readValue(taskUnitConfigJson, FeedRangeTaskUnit.class);
+                        } catch (JsonProcessingException e) {
+                            throw new IllegalArgumentException("Failed to parseFeedRangeTaskUnit", e);
+                        }
+                    })
+                    .collect(Collectors.toList());
+            }
+
+            return new ArrayList<>();
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException("Failed to parseFeedRangeTaskUnits", e);
+        }
+
+    }
+
+    private MetadataTaskUnit parseMetadataTaskUnit() {
+        String metadataTaskUnitConfig = this.getString(SOURCE_METADATA_TASK_UNIT);
+        if (!StringUtils.isEmpty(metadataTaskUnitConfig)) {
+            try {
+                return OBJECT_MAPPER.readValue(metadataTaskUnitConfig, MetadataTaskUnit.class);
+            } catch (JsonProcessingException e) {
+                throw new IllegalArgumentException("Failed to parseMetadataTaskUnit", e);
+            }
+        }
+
+        return null;
+    }
+
+    public static Map<String, String> getFeedRangeTaskUnitsConfigMap(List<FeedRangeTaskUnit> feedRangeTaskUnits) {
+        try {
+            Map<String, String> taskConfigMap = new HashMap<>();
+            taskConfigMap.put(
+                SOURCE_FEED_RANGE_TASK_UNITS,
+                OBJECT_MAPPER.writeValueAsString(
+                    feedRangeTaskUnits
+                        .stream()
+                        .map(taskUnit -> {
+                            try {
+                                return OBJECT_MAPPER.writeValueAsString(taskUnit);
+                            } catch (JsonProcessingException e) {
+                                throw new RuntimeException(e);
+                            }
+                        })
+                        .collect(Collectors.toList())
+                ));
+            return taskConfigMap;
+        } catch (JsonProcessingException e) {
+            throw new ConfigException("Failed to getFeedRangTaskUnitsConfigMap ", e);
```

Comment:
Why not RuntimeException? when we try to write an object to json it should never be due to config?

Reply:
make sense, changed to use RuntimeException